### PR TITLE
add support for Vector WithdrawHelper interface

### PR DESCRIPTION
**Pull request by geoknee**
* Requires some minor modifications to the vector `WithdrawHelper` interface (which I hope would be uncontroversial).
* includes an `executeExit` function to make payouts
* includes an example of how we can encode nitro guarantees in the new format. They will cause the `executeExit` function to revert (by design). A custom nitro `claim` function can make the appropriate payouts for guarantees, instead.

### CODE REVIEW

File: contracts/ExitFormat.sol [hunk 1]
```diff
@@ -63,74 +72,30 @@ library ExitFormat {
...
+    function executeExit(ExitFormat.SingleAssetExit[] memory exit) public {
+        for (uint256 i = 0; i < exit.length; i++) {
+            address asset = exit[i].asset;
+            for (uint256 j = 0; j < exit[i].allocations.length; j++) {
+                address payable destination =
+                    exit[i].allocations[j].destination;
+                uint256 amount = exit[i].allocations[j].amount;
+                address callTo = exit[i].allocations[j].callTo;
+                bytes memory data = exit[i].allocations[j].data;
+                if (asset == address(0)) {
```
Comment (andrewgordstewart) [inline contracts/ExitFormat.sol:89]: Can the `WithdrawHelper` handle this logic?

File: ts/coders.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
-import { AbiCoder, defaultAbiCoder, ParamType } from "@ethersproject/abi";
-import { abort } from "node:process";
+import { defaultAbiCoder, ParamType } from "@ethersproject/abi";
 import { Allocation, Exit } from "./types";
 
 export function encodeAllocations(allocation: Allocation) {
   return defaultAbiCoder.encode(
-    ["tuple(address destination, uint256 amount, bytes data)"],
+    ["tuple(address destination, uint256 amount,address callTo, bytes data)"],
```
Comment (NiloCK) [inline ts/coders.ts:6]:
```suggestion
    ["tuple(address destination, uint256 amount, address callTo, bytes data)"],
```

**Review by NiloCK — APPROVED** [review]
LGTM

File: contracts/ExitFormat.sol (hunk 1 above)
Comment (geoknee) [reply contracts/ExitFormat.sol:89]: A `WithdrawHelper` (as I understand it) may be used to forward ETH or tokens to a third party contract (as well as performing other arbitrary logic). So as long as the withdraw helper is called with a `CALL` (not `DELEGATECALL`), the funds need to move first into the `WithdrawHelper`. 

To maintain compatibility with the current `WithdrawHelpers`, we can stick with this for now.

**Pull request merged**